lib/katello/engine.rb

```diff
@@ -4,11 +4,13 @@ class Engine < ::Rails::Engine
 
     initializer 'katello.selective_params_parser', :before => :build_middleware_stack do |app|
       require 'katello/params_parser_wrapper'
-      app.middleware.swap(
-        ActionDispatch::ParamsParser,
-        Katello::ParamsParserWrapper,
-        ->(env) { env['PATH_INFO'] =~ /consumers/ && env['PATH_INFO'] =~ /profile|packages/ }
-      )
+      if Rails::VERSION::MAJOR == 4
+        app.middleware.swap(
+          ActionDispatch::ParamsParser,
+          Katello::ParamsParserWrapper,
+          ->(env) { env['PATH_INFO'] =~ /consumers/ && env['PATH_INFO'] =~ /profile|packages/ }
+        )
+      end
```

Comment:
> I believe its used to prevent parsing of the body as json to conserve memory.

Indeed, I want to prevent parsing the JSON to hash structure. We can pas the raw string to the dynflow action and parse it there.

> I think we would have to extend ActionDispatch::Request::Utils::ParamEncoder now to do it 

We don't have to do it (at least from what I saw), Rails 5 parses params through controller's helper methods as far as I know.
We can inherit [candlepin_dynflow_proxy_controller](https://github.com/Katello/katello/blob/master/app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb) from [API controller](http://api.rubyonrails.org/classes/ActionController/API.html) or even [Metal](http://api.rubyonrails.org/classes/ActionController/Metal.html).